rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -73,6 +73,92 @@ docker login --username ${cred[1]} --password ${cred[2]} ${cred[0]} \
   || { echo 'Authorized failed' ; exit 1; }
 {{/ jobData.authFile }}
 
+
+# Get container host list from framework launcher webservice
+function get_host_list()
+{
+  delay_time=1
+  while true; do
+    taskrole_statuses=$(curl -i -L -H "Accept: application/json" {{{ frameworkInfoWebhdfsUri }}} | jq -r ".aggregatedFrameworkStatus.aggregatedTaskRoleStatuses")
+    jq -re ".[].taskStatuses.taskStatusArray[].containerIp" <<< $(echo $taskrole_statuses) > /dev/null
+    if [ $? -eq 0 ]; then
+      break
+    fi
+    sleep ${delay_time}m
```

Comment:
>sleep ${delay_time}m [](start = 4, length = 20)

Since it is fetched from HDFS, we can reduce the interval to check. Such as to 20s, 40s, 60s....

Reply:
updated